apiserver/facade/facadetest/context.go

```diff
@@ -200,6 +201,9 @@ func (c ModelContext) HTTPClient(purpose corehttp.Purpose) (facade.HTTPClient, e
 
 // DomainServices implements facade.ModelContext.
 func (c ModelContext) DomainServices() services.DomainServices {
+	if c.DomainServices_ == nil {
+		return domainservicestesting.NewPlaceholderDomainServices()
+	}
```

Comment:
I think we should just delete this and force people to do the right thing.

Reply:
Have hunted down the usages and made the changes. Placeholder gone.